### PR TITLE
Update unit tests for LessonService

### LINKED ISSUE #369: Update unit tests for LessonService
Break huge tests into smaller ones (by method)

### PR DESCRIPTION
**Pull request by zelya2021**
Break huge tests into smaller ones (by method)

### CODE REVIEW

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 1]
```diff
@@ -11,20 +11,87 @@
...
+            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(_mentorRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(_studentGroupRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.VisitRepository).Returns(_visitRepositoryMock.Object);
+
+            _currentUserServiceMock = GetCurrentUserAsExistingStudent();
         }
 
         [Fact]
-        public async Task CreateLesson_ReturnLessonDTo_NotEmpty()
+        public async Task CreateLesson_LessonDTO_ShouldBeNotNull()
```
Comment (mahotskuy) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: The convention of unit tests should be next MethodName_State_ShoudlExpectedResult.
Please change LessonDTO to ValidDataPassed

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 2]
```diff
@@ -57,74 +124,111 @@ public async Task CreateLesson_ReturnLessonDTo_NotEmpty()
...
+            };
+            #endregion
 
-            _currentUserServiceMock = GetCurrentUserAsExistingStudent();
+            #region MOCK
+            var mentorReposutoryMockWrong = new Mock<IMentorRepository>();
+            _mentorRepositoryMock.Setup(x => x.GetMentorByIdAsync(mentor.Id)).ReturnsAsync(mentor);
+            mentorReposutoryMockWrong.Setup(x => x.GetMentorByIdAsync(mentorWrong.Id)).ReturnsAsync(default(Mentor));
+
+            MockEntities(studentGroup);
```
Comment (mahotskuy) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: You can call this method in the constructor. As Mock is an object, then can be configured after in any point before returning.

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 3]
```diff
@@ -57,74 +124,111 @@ public async Task CreateLesson_ReturnLessonDTo_NotEmpty()
...
             var lessonService = new LessonService(
-                unitOfWork: _unitOfWorkMock.Object, 
+               unitOfWork: _unitOfWorkMock.Object,
+               mapper: _mapper,
+               currentUserService: _currentUserServiceMock.Object);
+
+            #endregion
+            //Act 
+            var result = await lessonService.CreateLessonAsync(createLessonDto);
+            var resultWihtWrongMentor = await lessonService.CreateLessonAsync(createLessonDtoWrongMentor);
```
Comment (mahotskuy) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs:175]: As we test method and one specific state then we have to have only one ACT in each test.
Please write new test for testing expected MentorId.

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 4]
```diff
@@ -57,74 +124,111 @@ public async Task CreateLesson_ReturnLessonDTo_NotEmpty()
...
+            var lessonService = new LessonService(
+                unitOfWork: _unitOfWorkMock.Object,
                 mapper: _mapper,
                 currentUserService: _currentUserServiceMock.Object);
 
-            //Act
-            var result = await lessonService.CreateLessonAsync(createLessonDTO);
+            #endregion
+            //Act 
+            var result = await lessonService.CreateLessonAsync(createLessonDto);
```
Comment (mahotskuy) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: Test should have one act

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 1 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: done

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 2 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: done

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 3 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs:175]: done

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 4 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: Ok, fixed

**Review by mahotskuy — APPROVED** [review]

**Review by TetianaDomanska-dev — APPROVED** [review]

**Review by andy-dendy-repo — APPROVED** [review]

**Review by Serhii-Dudar — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 5]
```diff
@@ -11,104 +11,120 @@
...
 
-            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(lessonRepositoryMock.Object);
-            _unitOfWorkMock.Setup(x => x.ThemeRepository).Returns(themeRepositoryMock.Object);
-            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorReposutoryMock.Object);
-            _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(studentGroupRepositoryMock.Object);
-            _unitOfWorkMock.Setup(x => x.VisitRepository).Returns(visitRepositoryMock.Object);
 
             _currentUserServiceMock = GetCurrentUserAsExistingStudent();
+        }
+        [Fact]
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]:
```suggestion
        }
        
        [Fact]
```
Adding 1 empty line between test methods

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 6]
```diff
@@ -119,56 +135,140 @@ public async Task CreateLesson_ReturnLessonDTo_NotEmpty()
...
+
+            //Assert
+            resultWihtWrongMentor.Error.Code.Should().BeEquivalentTo(ErrorCode.NotFound);
+        }
+
+        [Fact]
+        public async Task CreateLessonAsync_WrongLessonDate_ShouldReturnValidationError()
+        {
+            //Arrange
+            #region DATA
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs:185]: We don't really need these regions, do we?

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 7]
```diff
@@ -178,142 +278,58 @@ public async Task CreateLessonAsync()
                     Presence = false
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs:281]: > StudentId = 1

Please replace the magic number in the line above with a constant.

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 8]
```diff
@@ -178,142 +278,58 @@ public async Task CreateLessonAsync()
...
+            //Assert
+            resultWithWrongLessonVisit.Error.Code.Should().BeEquivalentTo(ErrorCode.NotFound);
+        }
+
+        [Fact]
+        public async Task CreateLessonAsync_EmptyLessonVisit_ShouldReturnValidationError()
+        {
+            //Arrange
+            #region DATA
+            Mentor mentor = new Mentor() { Id = 2 };
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: In some cases you're using constants, in others you don't
asdasddas

Please replace the magic number

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 9]
```diff
@@ -178,142 +278,58 @@ public async Task CreateLessonAsync()
...
-            mentorReposutoryMockWrong.Setup(x => x.GetMentorByIdAsync(mentorWrong.Id)).ReturnsAsync(default(Mentor));
-
-            var studentGroupRepositoryMock = new Mock<IStudentGroupRepository>();
-            studentGroupRepositoryMock.Setup(x => x.GetByIdAsync(studentGroup.Id)).ReturnsAsync(studentGroup);
-            studentGroupRepositoryMock.Setup(x => x.GetGroupStudentsIds(studentGroup.Id)).ReturnsAsync(new List<long?> { 11, 14 });
+            _mentorRepositoryMock.Setup(x => x.GetMentorByIdAsync(mentor.Id)).ReturnsAsync(mentor);
 
             var studentGroupRepositoryMockWrong = new Mock<IStudentGroupRepository>();
-            studentGroupRepositoryMockWrong.Setup(x => x.GetByIdAsync(studentGroupWrong.Id)).ReturnsAsync(default(StudentGroup));
             studentGroupRepositoryMockWrong.Setup(x => x.GetGroupStudentsIds(createLessonDtoWrongLessonVisitsStudent.StudentGroupId)).ReturnsAsync(new List<long?> { 11 });
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: Please replace the magic number inside the list with a constant

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 10]
```diff
@@ -119,56 +135,140 @@ public async Task CreateLesson_ReturnLessonDTo_NotEmpty()
...
+
+        [Fact]
+        public async Task CreateLessonAsync_WrongLessonVisitsWithoutStudent_ShouldReturnValidationError()
+        {
+            //Arrange
+            #region DATA
+
+            Mentor mentor = new Mentor() { Id = mentorId };
+
+            StudentGroup studentGroupWrong = new StudentGroup() { Id = 100 };
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: Please replace the magic number with a constant

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 5 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: done

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 6 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs:185]: Why not?

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 7 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs:281]: done

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 8 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: done

**Pull request merged**